src/python_testing/TC_EWATERHTR_2_1.py

```diff
@@ -0,0 +1,120 @@
+            TestStep("6", "TH reads TankPercentage attribute.",
+                     "DUT as Server replies with a percent value."),
+            TestStep("7", "TH reads BoostState attribute.",
+                     "DUT as Server replies with a BoostStateEnum (enum8) value."),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_1(self):
+
+        em_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.EM']
+        tp_supported = self.matter_test_config.global_test_params['PIXIT.EWATERHTR.TP']
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_greater(heaterTypes, 0,
+                               f"Unexpected HeaterTypes value - expected {heaterTypes} > 0")
+        asserts.assert_less_equal(heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kOther,
+                                  f"Unexpected HeaterTypes value - expected {heaterTypes} <= WaterHeaterTypeBitmap.kOther")
+
+        self.step("3")
+        heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
+        asserts.assert_greater(heatDemand, 0,
+                               f"Unexpected HeatDemand value - expected {heatDemand} > 0")
+        asserts.assert_less_equal(heatDemand, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterDemandBitmap.kOther,
+                                  f"Unexpected HeatDemand value - expected {heatDemand} <= WaterHeaterDemandBitmap.kOther")
+
+        self.step("4")
+        if em_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="TankVolume")
+            asserts.assert_greater_equal(value, 0, f"Unexpected TankVolume value - expected {value} >= 0")
+        else:
+            logging.info("Skipping step 4 as PIXIT.EWATERHTR.EM not supported")
+
+        self.step("5")
+        if em_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="EstimatedHeatRequired")
+            asserts.assert_greater(value, 0, f"Unexpected EstimatedHeatRequired value - expected {value} > 0")
+        else:
+            logging.info("Skipping step 5 as PIXIT.EWATERHTR.EM not supported")
+
+        self.step("6")
+        if tp_supported:
+            value = await self.read_whm_attribute_expect_success(attribute="TankPercentage")
+            asserts.assert_greater_equal(value, 0, f"Unexpected TankPercentage value - expected {value} >= 0")
+            asserts.assert_less_equal(value, 100, f"Unexpected TankPercentage value - expected {value} <= 100")
+        else:
+            logging.info("Skipping step 6 as PIXIT.EWATERHTR.TP not supported")
+
+        self.step("7")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
```

Comment:
The test plan here expects:

> a BoostStateEnum (enum8) value.
But here we check for kInactive, we should make the two coherent.

Reply:
Missed that one - will add to the follow up PR to fix.
https://github.com/project-chip/connectedhomeip/issues/34565

Reply:
Added fix to check it is <= kInactive.